### PR TITLE
Style product

### PR DESCRIPTION
**Pull request by mikeyrayvon**

### CODE REVIEW

File: src/css/site.styl [hunk 1]
```diff
@@ -48,6 +52,25 @@ img
 .desktop-only
   display: none !important
 
+input, select, a
+  &:focus
+    outline: none
+    color: $color-red
+    img
+      box-shadow: 0 0 40px -5px $color-red
```
Comment (varffs) [inline src/css/site.styl:60]: doesnt need to be now but seems like this could be a mixin for use on other things?

File: src/js/main.js [hunk 2]
```diff
@@ -20,7 +20,11 @@ Site = {
     var _this = this;
 
     if ($('#product-gallery').length) {
-      _this.Product.setGalleryWidth();
+
+      // debounce gallery
```
Comment (varffs) [inline src/js/main.js]: this isn't a debounce tho it will stack the calls

File: src/js/main.js [hunk 3]
```diff
@@ -41,22 +45,94 @@ Site.Product = {
...
   },
 
-  setGalleryWidth: function() {
+  setGalleryDimensions: function() {
+    var _this = this;
     var galleryWidth = 0;
+    var windowWidth = $(window).width();
 
-    $('.product-gallery-item').each(function() {
+    $.each($('.product-gallery-item'), function() {
```
Comment (varffs) [inline src/js/main.js]: why did you switch the each here? The selector is jquery so can just each itself

File: src/js/main.js [hunk 4]
```diff
@@ -41,22 +45,94 @@ Site.Product = {
...
+    $('#product-gallery-row').width(galleryWidth + 1);
+
+    if (galleryWidth > windowWidth) {
+      var galleryContainerWidth = galleryWidth + (galleryWidth - windowWidth);
+
+      $('#product-gallery')
+        .width(galleryContainerWidth - 1)
+        .css('left', ((windowWidth / 2) - (galleryContainerWidth / 2)) + 'px');
+
+      $('#product-gallery-row').css('left', ((galleryContainerWidth / 2) - (galleryWidth / 2)) + 'px');
```
Comment (varffs) [inline src/js/main.js]: this seems to be the same value as L78 so could jube be a `var`?

File: src/js/main.js [hunk 5]
```diff
@@ -41,22 +45,94 @@ Site.Product = {
...
+
```
Comment (varffs) [inline src/js/main.js]: some code comments on what and why on this would be good here

File: src/js/main.js [hunk 6]
```diff
@@ -41,22 +45,94 @@ Site.Product = {
...
+      constrainTo: 'parent',
+      place: false,
+      useCSSTranslation: false
+    }).css('cursor', 'ew-resize');
+
+  },
+
+  unbindGalleryDrag: function() {
+    $.pep.unbind($('#product-gallery-row'));
+    $('#product-gallery-row').css('cursor', 'default');
```
Comment (varffs) [inline src/js/main.js]: is this reset when wanted back to the move arrow?

File: src/js/main.js [hunk 7]
```diff
@@ -41,22 +45,94 @@ Site.Product = {
...
+  },
+
+  selectCallback: function(variant, selector) {
+    if (variant && variant.available === true) {
+      $('#product-add-holder').removeClass('out-of-stock');
+      $('#add').removeAttr('disabled');
+    }
```
Comment (varffs) [inline src/js/main.js]: weird formatting

**Review by varffs — CHANGES_REQUESTED** [review]
A few bits in the inline comments. Main requests are debounce improvement and code comments for gallery.

However looking in my dev tools the gallery output seems a little odd. It seems to be working but:

screen shot 2017-03-23 at 19 18 32

you can see in that screenshot that the left value of the overall parent element is a negative value even when the gallery is fully scrolled to the start. That seems strange to me. Maybe its all fine but I guess I want to know how it works before I can know that

File: src/js/main.js (hunk 4 above)
Comment (mikeyrayvon) [reply src/js/main.js]: its not the same value tho

File: src/js/main.js (hunk 6 above)
Comment (mikeyrayvon) [reply src/js/main.js]: yea in the bind

File: src/css/site.styl (hunk 1 above)
Comment (mikeyrayvon) [reply src/css/site.styl:60]: im going to define the hover states in the next branch. i'll do that then

**Review by varffs — APPROVED** [review]

**Pull request merged**